test/Npgsql.Tests/SecurityTests.cs

```diff
@@ -240,10 +240,23 @@ public void SslMode_Require_throws_without_TSC()
             SslMode = SslMode.Require
         };
 
-        var ex = Assert.ThrowsAsync<NpgsqlException>(async () => await OpenConnectionAsync(csb))!;
-        Assert.That(ex.Message, Is.EqualTo("To validate server certificates, please use VerifyFull or VerifyCA instead of Require. " +
-                                           "To disable validation, explicitly set 'Trust Server Certificate' to true. " +
-                                           "See https://www.npgsql.org/doc/release-notes/6.0.html for more details."));
+        var ex = Assert.ThrowsAsync<ArgumentException>(async () => await OpenConnectionAsync(csb))!;
+        Assert.That(ex.Message, Is.EqualTo(NpgsqlStrings.CannotUseSslModeRequireWithoutTrustServerCertificate));
+    }
+
+    [Test]
+    public async Task SslMode_Require_with_callback_without_TSC()
+    {
+        var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+        {
+            SslMode = SslMode.Require,
+            Pooling = false
```

Comment:
I think we should explicitly set here `TrustServerCertificate` to `false` since the test is named `without_TSC`.

Reply:
It's the default (and should always stay the default), but sure - I'll add it.